$OpenBSD$

header fix
replace ftime() with gettimeofday() like emulators/fs-uae
fix relative path of header files

Index: src/ints/bios.cpp
--- src/ints/bios.cpp.orig
+++ src/ints/bios.cpp
@@ -42,7 +42,6 @@
 extern bool PS1AudioCard;
 #include "parport.h"
 #include <time.h>
-#include <sys/timeb.h>
 
 #if defined(WIN32) && !defined(S_ISREG)
 # define S_ISREG(x) ((x & S_IFREG) == S_IFREG)
@@ -3175,11 +3174,11 @@ static Bitu INT11_Handler(void) {
 
 static void BIOS_HostTimeSync() {
 	/* Setup time and date */
-	struct timeb timebuffer;
-	ftime(&timebuffer);
+	struct timeval timebuffer;
+	gettimeofday(&timebuffer, NULL);
 	
 	struct tm *loctime;
-	loctime = localtime (&timebuffer.time);
+	loctime = localtime (&timebuffer.tv_sec);
 
 	/*
 	loctime->tm_hour = 23;
@@ -3198,7 +3197,7 @@ static void BIOS_HostTimeSync() {
 		loctime->tm_hour*3600*1000+
 		loctime->tm_min*60*1000+
 		loctime->tm_sec*1000+
-		timebuffer.millitm))*(((double)PIT_TICK_RATE/65536.0)/1000.0));
+		timebuffer.tv_usec / 1000))*(((double)PIT_TICK_RATE/65536.0)/1000.0));
 	mem_writed(BIOS_TIMER,ticks);
 }
 
@@ -4504,8 +4503,8 @@ bool AdapterROM_Read(Bitu address,unsigned long *size)
 	return false;
 }
 
-#include "src/gui/dosbox.vga16.bmp.h"
-#include "src/gui/dosbox.cga640.bmp.h"
+#include "../src/gui/dosbox.vga16.bmp.h"
+#include "../src/gui/dosbox.cga640.bmp.h"
 
 void DrawDOSBoxLogoCGA6(unsigned int x,unsigned int y) {
 	unsigned char *s = dosbox_cga640_bmp;
